### DISCUSSION TITLE
SHAP for Deep text classificaion

### DISCUSSION THREAD

**Discussion by tejas0809** [shap/shap#1478] [General] [answered]
Hi, I am working on using SHAP for a sentiment classification model on textual data in PyTorch, where I plan to use SHAP values for features and average those over words, in order to get word-level ratings for a vocabulary. I am unsure of how should I pick a background for my DeepExplainer. Can I take a random subset of tokens from my train data? Also, I came across shap TokenMasker, and was unsure about how to use it. Any leads would be great. 
Also, thanks for the awesome explanation for Shap in the docs. 
Thanks!

**Comment by slundberg** [comment, marked as answer]
Hey! This is a great use case for the new Text masker (which is not documented really yet because it is still being finalized). You can clone master though as use it for just what you are after by following the text plot examples: https://nbviewer.jupyter.org/github/slundberg/shap/blob/master/notebooks/plots/text.ipynb

Just keep in mind that we are still putting the finishing touches on that code so it might change a little in the future.

**Reply by tejas0809** [reply]
Thanks, I will look into it! 😄

**Reply by martin-koder** [reply]
I'm trying something similar but the above link is broken. I can't get SHAP to work with my torchtext model at all. I can't identify where the source code is either in order to hack into it and amend - links from the documentation seem broken and I can't seem to find them on GitHub. My torchtext model uses offsets and the explainer is not set up for that, but where is the source code to hack it?

**Comment by y12uc231** [comment]
The notebook link mentioned above is broken. Has there been any progress on this thread yet? I am also trying to get shap values for a torch based text classifier.
